backend/api/gitops/webhook.go

```diff
@@ -289,3 +279,60 @@ func (s *Service) createIssueFromPRInfo(ctx context.Context, project *store.Proj
 	}
 	return nil
 }
+
+func (s *Service) listDatabases(ctx context.Context, project *store.ProjectMessage, vcsConnector *store.VCSConnectorMessage) ([]*store.DatabaseMessage, error) {
+	if vcsConnector.Payload.DatabaseGroup != "" {
+		return listMatchedDatabasesInGroup(ctx, s.store, project, vcsConnector.Payload.DatabaseGroup)
+	}
+
+	databases, err := s.store.ListDatabases(ctx, &store.FindDatabaseMessage{ProjectID: &project.ResourceID})
+	if err != nil {
+		return nil, err
+	}
+	environments, err := s.store.ListEnvironmentV2(ctx, &store.FindEnvironmentMessage{})
+	if err != nil {
+		return nil, err
+	}
+	environmentOrders := make(map[string]int32)
+	for _, environment := range environments {
+		environmentOrders[environment.ResourceID] = environment.Order
+	}
+	sort.Slice(databases, func(i, j int) bool {
+		return environmentOrders[databases[i].EffectiveEnvironmentID] < environmentOrders[databases[j].EffectiveEnvironmentID]
+	})
+
+	return databases, nil
+}
+
+func listMatchedDatabasesInGroup(ctx context.Context, stores *store.Store, project *store.ProjectMessage, databaseGroupName string) ([]*store.DatabaseMessage, error) {
+	projectID, databaseGroupID, err := common.GetProjectIDDatabaseGroupID(databaseGroupName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get project id and database group id from target %q", databaseGroupName)
+	}
+	if projectID != project.ResourceID {
+		return nil, errors.Errorf("project id %q in target %q does not match project id %q in plan config", projectID, databaseGroupName, project.ResourceID)
+	}
+
+	databaseGroup, err := stores.GetDatabaseGroup(ctx, &store.FindDatabaseGroupMessage{ProjectUID: &project.UID, ResourceID: &databaseGroupID})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get database group %q", databaseGroupID)
+	}
+	if databaseGroup == nil {
+		return nil, errors.Errorf("database group %q not found", databaseGroupID)
+	}
+
+	allDatabases, err := stores.ListDatabases(ctx, &store.FindDatabaseMessage{ProjectID: &project.ResourceID})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to list databases for project %q", project.ResourceID)
+	}
+
+	matchedDatabases, _, err := utils.GetMatchedAndUnmatchedDatabasesInDatabaseGroup(ctx, databaseGroup, allDatabases)
```

Comment:
We can use database group target in the plan spec.

https://github.com/bytebase/bytebase/blob/d5e0ecbd2e8c2dc443e1eeecd5f489fb3e0bcdb3/proto/v1/rollout_service.proto#L229

Reply:
You mean use `target` filed instead of `database_group` in the [store](https://github.com/bytebase/bytebase/pull/11458/files#diff-d2afe533d0a7c3e6c407977be13d030168fed31413f89bf39854f08af9a3ceecR30) and [API](https://github.com/bytebase/bytebase/pull/11458/files#diff-058457964c2f8d04ce54747f31e18d4c1b377c5abee5fb5db03df80e15b188abR163), and the target could be database, database group or deployment config?

Reply:
I mean we can avoid computing the databases in the database group by setting the plan target to "projects/{project}/databaseGroups/{databaseGroup}" i.e. the database_group field in the vcs connector.

Reply:
Got it